tracer/src/Datadog.Trace/Propagators/B3SingleHeaderContextPropagator.cs

```diff
@@ -106,12 +118,23 @@ private B3SingleHeaderContextPropagator()
                     return false;
                 }
 
-                var traceId = rawTraceId.Length == 32 ?
-                                  ParseUtility.ParseFromHexOrDefault(rawTraceId.Substring(16)) :
-                                  ParseUtility.ParseFromHexOrDefault(rawTraceId);
-                var parentId = ParseUtility.ParseFromHexOrDefault(rawSpanId);
-                var samplingPriority = rawSampled == '1' ? 1 : 0;
+                ulong traceId;
+
+                var success = rawTraceId.Length == 32 ?
+                                  HexString.TryDecodeToUInt64(rawTraceId.Substring(16), out traceId) :
+                                  HexString.TryDecodeToUInt64(rawTraceId, out traceId);
 
+                if (!success || traceId == 0)
```

Comment:
Note to reviewers: there was a bug here because we were not checking for `traceId == 0`.